terrarium/dl_gitmanager/dl_gitmanager/git_manager.py

```diff
@@ -114,23 +114,35 @@ def _iter_list_diffs(
             for _, diff_item in submodule_manager._iter_list_diffs(commits=sm_commits_for_all_commits):
                 yield submodule_base_path, diff_item
 
-    def _collect_paths_from_diffs(self, diff_iterable: Iterable[tuple[Path, Diff]]) -> list[str]:
+    def _collect_paths_from_diffs(
+        self, diff_iterable: Iterable[tuple[Path, Diff]], without_deleted: bool = False
+    ) -> list[str]:
         result: set[str] = set()
         for base_path, diff_item in diff_iterable:
-            if diff_item.a_path:
-                result.add(str(base_path / diff_item.a_path))
-            if diff_item.b_path:
-                result.add(str(base_path / diff_item.b_path))
+            for diff_path in (diff_item.a_path, diff_item.b_path):
```

Comment:
Better to do this in a separate cycle - first we collect and deduplicate all the paths, and then we apply filters.
It would be a bit more flexible and would save some CPU time if there is a lot of commits for the same paths.